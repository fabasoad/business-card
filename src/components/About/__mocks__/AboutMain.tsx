import * as React from 'react'

export default function AboutMain() {
  return (
    <div data-testid="AboutMain"></div>
  )
}
