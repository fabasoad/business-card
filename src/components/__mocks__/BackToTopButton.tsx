import * as React from 'react'

export default function BackToTopButton() {
  return (
    <div data-testid="BackToTopButton"></div>
  )
}
