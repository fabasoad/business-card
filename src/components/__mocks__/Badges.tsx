import * as React from 'react'

export default function Badges() {
  return (
    <div data-testid="Badges"></div>
  )
}
