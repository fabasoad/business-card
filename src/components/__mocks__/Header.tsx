import * as React from 'react'

export default function Header() {
  return (
    <div data-testid="Header"></div>
  )
}
