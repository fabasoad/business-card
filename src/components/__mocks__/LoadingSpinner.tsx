import * as React from 'react'
import { LoadingSpinnerProps } from '../LoadingSpinner'

export default function LoadingSpinner({ background, color }: LoadingSpinnerProps) {
  return (
    <div data-testid={`LoadingSpinner-${background}-${color}`}></div>
  )
}
