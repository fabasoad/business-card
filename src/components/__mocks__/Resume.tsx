import * as React from 'react'

export default function Resume() {
  return (
    <div data-testid="Resume"></div>
  )
}
