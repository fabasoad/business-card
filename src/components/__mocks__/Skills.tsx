import * as React from 'react'

export default function Skills() {
  return (
    <div data-testid="Skills"></div>
  )
}
