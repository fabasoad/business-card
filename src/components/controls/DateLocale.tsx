import * as React from 'react'
import { useTranslation } from 'react-i18next'
import { connect } from 'react-redux'
import { ThunkDispatch } from 'redux-thunk'
import { Locale, AppActions } from '../../store/locale/types'
import { AppState } from '../../store/configureStore'
import DigitConverter from '../../scripts/DigitConverter'

interface DateProps {
  className?: string
  converter?: DigitConverter
  month?: number
  year: number
}

type Props = DateProps & LinkDispatchProps & LinkStateProps

export function DateLocale(props: Props) {
  const { t } = useTranslation()

  const date: string = props.locale.code === 'jp'
    ? props.converter.toDoubleByte(props.year.toString()) + t('business-card-year-singular') + (props.month ? t(`business-card-month-${props.month}`) : '')
    : (props.month ? t(`business-card-month-${props.month}`) + ' ' : '') + props.year

  return (
    <div className={props.className || ''}>{date}</div>
  )
}

DateLocale.defaultProps = {
  converter: new DigitConverter()
}

interface LinkStateProps {
  locale: Locale
}

interface LinkDispatchProps {
}

const mapStateToProps = (
  state: AppState,
  ownProps: DateProps
): LinkStateProps => ({
  locale: state.locale
})

const mapDispatchToProps = (
  dispatch: ThunkDispatch<any, any, AppActions>,
  ownProps: DateProps
): LinkDispatchProps => ({
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(DateLocale)
