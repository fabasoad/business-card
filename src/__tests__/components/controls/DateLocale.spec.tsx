/// <reference types="jest" />
import * as React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import { DateLocale } from '../../../components/controls/DateLocale'
import { useTranslation } from '../../__mocks__/react-i18next'

let tMock

beforeAll(() => {
  tMock = useTranslation().t
})

interface DateLocaleFixture {
  title: string
  locale: string
  expect: (toDoubleByteSpy: jest.Mock<string, string[]>, year: number, month?: number) => void
}

new Array<DateLocaleFixture>({
  title: 'JP',
  locale: 'jp',
  expect: (toDoubleByteSpy, year, month) => {
    expect(tMock).toHaveBeenCalledTimes(2)
    expect(tMock).toHaveBeenNthCalledWith(1, 'business-card-year-singular')
    expect(tMock).toHaveBeenNthCalledWith(2, `business-card-month-${month}`)
    expect(toDoubleByteSpy).toHaveBeenCalledWith(year.toString())
  }
}, {
  title: 'non-JP',
  locale: 'gb',
  expect: (toDoubleByteSpy, year, month) => {
    expect(tMock).toHaveBeenCalledTimes(1)
    expect(tMock).toHaveBeenCalledWith(`business-card-month-${month}`)
    expect(toDoubleByteSpy).not.toHaveBeenCalled()
  }
}).forEach((f) =>
  test(`should render DateLocale with month in ${f.title} locale correctly`, () => {
    const month = 1
    const year = 2019
    tMock.mockImplementation((k) => `${f.locale}-${k}`)
    const toDoubleByteSpy: jest.Mock<string, string[]> = jest.fn((k) => k)
    const wrapper: ShallowWrapper = shallow(
      <DateLocale
        converter={{
          _convert: null,
          toDoubleByte: toDoubleByteSpy,
          toSingleByte: null
        }}
        locale={{
          code: f.locale,
          title: ''
        }}
        month={month}
        year={year}
      />
    )
    expect(wrapper).toMatchSnapshot()
    f.expect(toDoubleByteSpy, year, month)
  })
)

new Array<DateLocaleFixture>({
  title: 'JP',
  locale: 'jp',
  expect: (toDoubleByteSpy, year) => {
    expect(tMock).toHaveBeenCalledTimes(1)
    expect(tMock).toHaveBeenCalledWith('business-card-year-singular')
    expect(tMock).not.toHaveBeenCalledWith(expect.stringMatching(/^business-card-month-.*$/))
    expect(toDoubleByteSpy).toHaveBeenCalledWith(year.toString())
  }
}, {
  title: 'non-JP',
  locale: 'gb',
  expect: (toDoubleByteSpy, year) => {
    expect(tMock).not.toHaveBeenCalled()
    expect(toDoubleByteSpy).not.toHaveBeenCalled()
  }
}).forEach((f) =>
  test(`should render DateLocale without month in ${f.title} locale correctly`, () => {
    const year = 2019
    tMock.mockImplementation((k) => `${f.locale}-${k}`)
    const toDoubleByteSpy = jest.fn((k) => k)
    const wrapper: ShallowWrapper = shallow(
      <DateLocale
        converter={{
          _convert: null,
          toDoubleByte: toDoubleByteSpy,
          toSingleByte: null
        }}
        locale={{
          code: f.locale,
          title: ''
        }}
        year={year}
      />
    )
    expect(wrapper).toMatchSnapshot()
    f.expect(toDoubleByteSpy, year)
  })
)
