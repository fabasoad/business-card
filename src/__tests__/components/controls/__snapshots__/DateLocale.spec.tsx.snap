// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DateLocale with month in JP locale correctly 1`] = `
<div
  className=""
>
  2019jp-business-card-year-singularjp-business-card-month-1
</div>
`;

exports[`should render DateLocale with month in non-JP locale correctly 1`] = `
<div
  className=""
>
  gb-business-card-month-1 2019
</div>
`;

exports[`should render DateLocale without month in JP locale correctly 1`] = `
<div
  className=""
>
  2019jp-business-card-year-singular
</div>
`;

exports[`should render DateLocale without month in non-JP locale correctly 1`] = `
<div
  className=""
>
  2019
</div>
`;
